Library/ViewModels/SearchViewModel.swift

```diff
@@ -212,27 +220,18 @@ public final class SearchViewModel: SearchViewModelType, SearchViewModelInputs,
       .filter { _, page in page == 1 }
       .map(first)
 
-    self.goToProject = Signal.combineLatest(self.projects, query)
-      .takePairWhen(self.tappedProjectProperty.signal.skipNil())
-      .map { projectsAndQuery, tappedProject in
-        let (projects, query) = projectsAndQuery
-
-        return (tappedProject, projects, refTag(query: query, projects: projects, project: tappedProject))
-      }
-
-    // Tracking
+    let searchResultsCount = Signal.merge(
+      searchResults.filter { $0 == 0 },
+      searchResults.takeWhen(firstPageResults)
+    )
 
     Signal.combineLatest(query, requestFirstPageWith)
-      .takePairWhen(firstPageResults)
+      .takePairWhen(searchResultsCount)
       .map(unpack)
       .filter { query, _, _ in !query.isEmpty }
-      .observeValues { query, params, hasResults in
-        AppEnvironment.current.ksrAnalytics.trackSearchResults(
-          query: query,
-          params: params,
-          refTag: .search,
-          hasResults: hasResults
-        )
+      .observeValues { _, params, stats in
+        AppEnvironment.current.ksrAnalytics
+          .trackProjectSearchView(params: params, results: stats)
```

Comment:
I'm a little confused here. We're tracking the same event here and on line 211 but the logic is different. Is 211 necessary? Also the `searchResultsCount` property is very similar to `searchResults` above. Is this something we can simplify/re-factor?

Reply:
Yes there are two different logic and we are meant to use one event for both scenarios. They were previously tracked with two separate events. Here are the current two scenarios `Page Viewed (search)` (when user visits the page) and `Page Viewed Search Term` (when user enters a search term and gets a response).